nbextensions/appCell2/widgets/appCellWidget.js

```diff
@@ -1478,88 +1406,52 @@ define(
-                            // Thus we can just emit the state of the current job globally
-                            // on the cell bus for those widgets interested.
-                            cellBus.emit('job-state', {
-                                jobState: newJobState,
-                            });
-                        } else {
-                            cellBus.emit('job-state-updated', {
-                                jobId: newJobState.job_id,
-                            });
-                        }
-
-                        model.setItem('exec.jobStateUpdated', new Date().getTime());
+                jobListeners.push(
+                    runtime.bus().listen({
+                        channel: {
+                            jobId,
+                        },
+                        key: {
+                            type: 'job-status',
+                        },
+                        handle: doJobStatus,
+                    })
+                );
 
-                        updateFromJobState(newJobState, forceRender);
-                    },
+                runtime.bus().emit('request-job-updates-start', {
+                    jobId,
                 });
-                jobListeners.push(ev);
+            }
 
-                ev = runtime.bus().listen({
-                    channel: {
-                        jobId,
-                    },
-                    key: {
-                        type: 'job-canceled',
-                    },
-                    handle: function () {
-                        //  reset the cell into edit mode
-                        const state = fsm.getCurrentState();
-                        if (state.state.mode === 'editing') {
-                            console.warn('in edit mode, so not resetting ui');
-                            return;
-                        }
-                        resetToEditMode();
-                    },
-                });
-                jobListeners.push(ev);
+            function doJobStatus(message) {
+                const existingState = model.getItem('exec.jobState'),
+                    newJobState = message.jobState,
+                    { outputWidgetInfo } = message,
+                    forceRender =
+                        !Jobs.isValidJobStateObject(existingState) &&
+                        Jobs.isValidJobStateObject(newJobState);
+                if (!existingState || !_.isEqual(existingState, newJobState)) {
+                    model.setItem('exec.jobState', newJobState);
+                    if (outputWidgetInfo) {
+                        model.setItem('exec.outputWidgetInfo', outputWidgetInfo);
+                    }
 
-                ev = runtime.bus().listen({
-                    channel: {
-                        jobId,
-                    },
-                    key: {
-                        type: 'job-does-not-exist',
-                    },
```

Comment:
this message type has been removed; now handled by the doJobStatus function -- see the action for status = `does_not_exist`